### PR TITLE
Simplified Builder for Beeline

### PR DESCRIPTION
**Pull request by MarsianMan**
- Added BeelineBuilder to ease Beeline creation. 
  - See README in beeline-core's io.honeycomb.beeline.builder package for usage.
  - Use `debug(true)` to add DefaultResponseObserver
- Added support for BeelineBuilder for users of DefaultBeeline. 
- Added `Beeline.close()` to ensure Transport shutdown for users of the Beeline. This should improve beeline abstraction.
- Bumped version to snapshot
- Disabled doclint for Javadocs as references to unshaded Apache HTTP client classes caused build failure.

Build failure is due to requirement on upstream changes in libhoney. https://github.com/honeycombio/libhoney-java/pull/25

### CODE REVIEW

**Comment by MarsianMan** [comment]
FYI, @honeycombio/libhoney-java/#25 deprecates `io.honeycomb.beeline.spring.beans.DebugResponseObserver` as this functionality is now available to all users of LibHoney via `io.honeycomb.libhoney.DefaultDebugResponseObserver`.

File: pom.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <name>Beeline Java (Parent)</name>
     <artifactId>beeline-parent</artifactId>
     <groupId>io.honeycomb.beeline</groupId>
-    <version>1.1.0</version>
+    <version>1.1.1</version>
```
Comment (paulosman) [inline pom.xml:10]: Please revert this back to 1.1.0, we'll do the version bump in a separate PR.

File: beeline-spring-boot-starter/pom.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
     <description>Spring Boot Starter module to auto-configure Spring Boot with the Honeycomb Beeline for Java</description>
 
     <properties>
-        <beelineVersion>1.0.8</beelineVersion>
+        <beelineVersion>1.1.1</beelineVersion>
```
Comment (paulosman) [inline beeline-spring-boot-starter/pom.xml:19]: Let's keep this at 1.1.0

File: beeline-spring-boot-starter/pom.xml [hunk 3]
```diff
@@ -7,7 +7,7 @@
     <parent>
         <groupId>io.honeycomb.beeline</groupId>
         <artifactId>beeline-parent</artifactId>
-        <version>1.1.0</version>
+        <version>1.1.1</version>
```
Comment (paulosman) [inline beeline-spring-boot-starter/pom.xml:10]: Same here - let's keep it as 1.1.0, we'll do a separate PR for the version bump.

File: beeline-core/pom.xml [hunk 4]
```diff
@@ -7,7 +7,7 @@
     <parent>
         <groupId>io.honeycomb.beeline</groupId>
         <artifactId>beeline-parent</artifactId>
-        <version>1.1.0</version>
+        <version>1.1.1</version>
```
Comment (paulosman) [inline beeline-core/pom.xml:10]: Keep this 1.1.0 in this PR so the build works.

File: beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
+# Building Beelines with the BeelineBuilder
+BeelineBuilder is an attempt to create a unified API for Beelines to simplify user configuration between the Beeline and the HoneyClient.
```
Comment (paulosman) [inline beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md:2]: Let's take out "is an attempt to create" and just say "is a unified API ..."

File: beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+        // Call close at program termination to ensure all pending
+        // spans are sent.
+        beeline.close();
+    }
+}
+```
+
+## Using DefaultBeeline
+DefaultBeeline has a new `getInstance` method which takes BeelineBuilder and serviceName.
```
Comment (paulosman) [inline beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md:33]: remove the word "new"

File: beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+        // Call close at program termination to ensure all pending
+        // spans are sent.
+        beeline.close();
+    }
+}
+```
+
+## Simplified Proxy Configuration
+It is now much simpler for people to configure servers behind an http proxy. To add a proxy server that does not need authentication, simply point to the proxy service.
```
Comment (paulosman) [inline beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md:58]: This reads like a changelog or release notes. A person coming to this documentation won't necessarily know what was there before (and therefore won't see it as "Simplified"). Mind rewording to just state what needs to be done?

Comment (MarsianMan) [reply beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md]: Updated

File: beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md (hunk 6 above)
Comment (MarsianMan) [reply beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md]: Done.

File: beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md (hunk 5 above)
Comment (MarsianMan) [reply beeline-core/src/main/java/io/honeycomb/beeline/builder/README.md]: Updated.

File: pom.xml (hunk 1 above)
Comment (MarsianMan) [reply pom.xml:10]: As discussed in slack, I have not reverted this change.

File: beeline-spring-boot-starter/pom.xml (hunk 2 above)
Comment (MarsianMan) [reply beeline-spring-boot-starter/pom.xml:19]: As discussed in slack, I have not reverted this change.

File: beeline-spring-boot-starter/pom.xml (hunk 3 above)
Comment (MarsianMan) [reply beeline-spring-boot-starter/pom.xml:10]: As discussed in slack, I have not reverted this change.

File: beeline-core/pom.xml (hunk 4 above)
Comment (MarsianMan) [reply beeline-core/pom.xml:10]: As discussed in slack, I have not reverted this change.
